internal/exec/engine.go

```diff
@@ -76,7 +77,7 @@ func (e Engine) Run(stageName string) bool {
 
 	e.Logger.PushPrefix(stageName)
 	defer e.Logger.PopPrefix()
-	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
+	return stages.Get(stageName).Create(e.Logger, &e.client, e.Root, e.OEM).Run(config.Append(baseConfig, config.Append(e.OemBaseConfig, cfg)))
```

Comment:
You could create the fetcher very early on and just pass them to the stages. Maybe the stages themselves just get a reference to the engine.